spec/mutations/availability/create_availability_spec.rb

```diff
@@ -44,34 +48,74 @@ def query(status:)
-        expect(json['data']['createAvailability']['userId']).to eq(@user.id.to_s)
-        expect(json['data']['createAvailability']['startDateTime']).to eq(@start_dt.to_s)
-        expect(json['data']['createAvailability']['endDateTime']).to eq(@end_dt.to_s)
-        expect(json['data']['createAvailability']['status']).to eq('cancelled')
+        expect(json[:data][:createAvailability][:id]).to_not be(nil)
+        expect(json[:data][:createAvailability][:userId]).to eq(@user.id.to_s)
+        expect(json[:data][:createAvailability][:startDateTime]).to eq(@start_dt.to_s)
+        expect(json[:data][:createAvailability][:endDateTime]).to eq(@end_dt.to_s)
+        expect(json[:data][:createAvailability][:status]).to eq('cancelled')
+      end
+
+      it 'when an availability is created it tries to pair the user' do
+        japanese_learning_english = create(:user, id: 1)
+        start_dt = DateTime.new(2021, 1, 1, 14, 30).to_i
+        end_dt = DateTime.new(2021, 1, 1, 15, 30).to_i
+        english_learning_japanese = create(:user, id: 2)
+        english_learning_japanese_two = create(:user, id: 3)
+        create(:user_language, :native, language: @language1, user: japanese_learning_english)
+        create(:user_language, :target, language: @language2, user: japanese_learning_english)
+        create(:user_language, :native, language: @language2, user: english_learning_japanese)
+        create(:user_language, :target, language: @language1, user: english_learning_japanese)
+        create(:user_language, :native, language: @language2, user: english_learning_japanese_two)
+        create(:user_language, :target, language: @language1, user: english_learning_japanese_two)
+        english_learning_japanese.availabilities.create(start_date_time: DateTime.new(2021, 1, 1, 13, 30), end_date_time: DateTime.new(2021, 1, 1, 15, 45), status: 0)
+        english_learning_japanese_two.availabilities.create(start_date_time: DateTime.new(2021, 1, 1, 13, 30), end_date_time: DateTime.new(2021, 1, 1, 15, 45), status: 0)
+
+        post '/graphql', params: { query:
+          "mutation {
+          createAvailability(input: { params: {
+            userId: #{japanese_learning_english.id}
+            startDateTime: #{start_dt}
+            endDateTime: #{end_dt}
+          }}) {
+            id
+            userId
+            startDateTime
+            endDateTime
+            status
+          }
+        }" }
+
+        json = JSON.parse(response.body, symbolize_names: true)
+        expect(json[:data][:createAvailability][:id]).to_not be(nil)
+        expect(json[:data][:createAvailability][:userId]).to eq(japanese_learning_english.id.to_s)
+        expect(json[:data][:createAvailability][:startDateTime]).to eq(start_dt.to_s)
+        expect(json[:data][:createAvailability][:endDateTime]).to eq(end_dt.to_s)
+        expect(json[:data][:createAvailability][:status]).to eq('fulfilled')
+        
+        expect(Pairing.all.size).to eq(1)
```

Comment:
This is the main test for seeing if a pairing is automatically created and that the availability is set to fulfilled